src/aggregations/aggregation-builder.js

```diff
@@ -0,0 +1,64 @@
+import _ from 'lodash'
+import aggregations from './index'
+
+const AggregationBuilder = {
+  /**
+   * Apply a aggregation of a given type providing all the necessary arguments,
+   * passing these arguments directly to the specified aggregation builder.
+   * Merges existing aggregation(s) with the new aggregation.
+   *
+   * @param  {String}  type Name of the aggregation type.
+   * @param  {...args} args Arguments passed to aggregation builder. May include
+   *                        a nest function as the last item.
+   * @returns {AggregationBuilder} Builder object.
+   */
+  aggregation(type, ...args) {
+    let klass = aggregations[type]
+    let aggregation = Object.create(AggregationBuilder)
+
+    if (!klass) {
+      throw new TypeError(`Aggregation type ${type} not found.`)
+    }
+
+    // If last argument is a nesting function, remove it
+    // from `args` before building the aggregation
+    let nest = _.last(args)
+    args = _.isFunction(nest) ? _.initial(args) : args
+
+    // Mixin to assign newly built aggregation properties
+    // to the `AggregationBuilder`
+    _.extend(aggregation, klass(...args))
```

Comment:
Turns out, we could get away with just plain ol' `_.assign` here, so I changed it to that.